activerecord/lib/active_record/associations/has_many_association.rb

```diff
@@ -88,14 +88,8 @@ def cached_counter_attribute_name(reflection = reflection())
         end
 
         def update_counter(difference, reflection = reflection())
-          update_counter_in_database(difference, reflection)
-          update_counter_in_memory(difference, reflection)
-        end
-
-        def update_counter_in_database(difference, reflection = reflection())
```

Comment:
I simply moved method to the place where it is used: only in `has_many_through_association` but not `has_many_association`.